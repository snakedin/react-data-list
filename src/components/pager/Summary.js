import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withOptions } from "../../options/withOptions";

class Summary extends Component {

    static propTypes = {
        page: PropTypes.number,
        pageSize: PropTypes.number,
        totalCount: PropTypes.number
    };

    render() {

        const {
            page, pageSize, totalCount,
            options, options: { renderPageSummary }
        } = this.props;

        const begin = page*pageSize - pageSize + 1;
        let end = page*pageSize;

        if (end > totalCount) {
            end = totalCount;
        }

        if (typeof renderPageSummary === 'function') {
            return renderPageSummary(begin, end, totalCount);
        } else {
            return (
                <span dangerouslySetInnerHTML={{ __html: options.translate('pageSummary', { begin, end, totalCount }) }}/>
            );
        }
    }
}

export default withOptions(Summary);