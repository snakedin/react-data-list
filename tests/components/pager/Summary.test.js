import React from "react";
import { render } from "@testing-library/react";
import "../../mocks/withOptionsMock";
import Summary from "../../../src/components/pager/Summary";

it("pager/Summary render", () => {
  const { asFragment } = render(<Summary page={3} pageSize={15} totalCount={100} />);
  expect(asFragment()).toMatchSnapshot();
});
