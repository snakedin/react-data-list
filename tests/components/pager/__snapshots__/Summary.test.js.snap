// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pager/Summary render 1`] = `
<DocumentFragment>
  <span>
    Showing 
    <b>
      31-45
    </b>
     of 
    <b>
      100
    </b>
     items.
  </span>
</DocumentFragment>
`;
